docs/how-to/service-dependencies.md

```diff
@@ -1,11 +1,297 @@
 # How to manage service dependencies
 
-Pebble takes service dependencies into account when starting and stopping services. When Pebble starts a service, it also starts the services which that service depends on (configured with `required`). Conversely, when stopping a service, Pebble also stops services which depend on that service.
+If you are using Pebble to manage your services, chances are, you've got more than one service to manage.
+
+And when orchestrating more services, things can and will get trickier, because more often than not, those services depend on each other to function together, and managing dependencies is no mean feat.
+
+Let's have a look at a concrete example of service dependencies.
+
+## My web app
+
+Imagine we are running a web application, which has:
+
+- a database listening on port 3306
+- a backend server listening on port 8081, which talks to the database
+- a frontend server listening on port 8080, which talks to the backend
+
+I.E.:
+
+`frontend (8080) -> backend (8081) -> db (3306)`
+
+## Layers configuration
+
+For the above example, we may have Pebble layers configured as follows:
+
+`001-frontend.yaml`:
+
+```yaml
+services:
+  frontend-server:
+    override: replace
+    summary: frontend
+    command: python3 -m http.server 8080
+    startup: enabled
+```
+
+`002-backend.yaml`:
+
+```yaml
+services:
+  backend-server:
+    override: replace
+    summary: backend
+    command: python3 -m http.server 8081
+    startup: enabled
+```
+
+`003-database.yaml`:
+
+```yaml
+services:
+  database:
+    override: replace
+    summary: database
+    command: python3 -m http.server 3306
+    startup: enabled
+```
+
+
+```{note}
+We use Python's http module to mock the servers.
+
+Alternatively, you may choose to define those services in a single layer. For example:
```

Comment:
I agree that we don't need an alternative here. If there's a reason to present single YAML vs multiple YAML files for layers, it can be a separate guide / tutorial.

I agree that if a single YAML file does the job, then the sample provided in the beginning can be shown in 1 layer instead of multiple layers.